kvbc/include/categorization/blocks.h

```diff
@@ -53,116 +54,132 @@ struct Block {
   void setParentHash(const BlockDigest& parent_hash) {
     std::copy(parent_hash.begin(), parent_hash.end(), data.parent_digest.begin());
   }
+
   inline BlockId id() const { return data.block_id; }
 
-  static const detail::Buffer& serialize(const Block& block) {
-    static thread_local detail::Buffer output;
-    output.clear();
-    concord::kvbc::categorization::serialize(output, block.data);
-    return output;
-  }
+  static const detail::Buffer& serialize(const Block& block) { return detail::serialize(block.data); }
 
-  static Block deserialize(const detail::Buffer& input) {
+  template <typename T>
+  static Block deserialize(const T& input) {
     Block output;
-    concord::kvbc::categorization::deserialize(input, output.data);
+    detail::deserialize(input, output.data);
     return output;
   }
 
   // Generate block key from block ID
   // Using CMF for big endian
   static const detail::Buffer& generateKey(const BlockId block_id) {
-    static thread_local detail::Buffer output;
-    output.clear();
     BlockKey key{block_id};
-    // think about optimization
-    concord::kvbc::categorization::serialize(output, key);
-    return output;
+    return concord::kvbc::categorization::detail::serialize(key);
   }
 
   BlockData data;
 };
 
-//////////////////////////////////// RAW BLOCKS//////////////////////////////////////
+class BlockChain {
```

Comment:
I think this class may be put into the `detail` namespace as it is not supposed to be exposed to users.